### PR TITLE
net: lwm2m: Refactoring of lwm2m_engine.c, registry

### PR DESCRIPTION
**Pull request by otkulseng**
Created the file lwm2m_registry.c to include the
getters and setters of values and the creation
and deletion of all lwm2m objects, resources
and instances.

Signed-off-by: Ola Tangen Kulseng <ola.kulseng@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 1]
```diff
@@ -369,7 +367,7 @@ static int update_attrs(void *ref, struct notification_attrs *out)
 	return 0;
 }
 
-static void clear_attrs(void *ref)
+void clear_attrs(void *ref)
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_engine.c:370]: Please place the declaration into _lwm2m_engine.h_ and document it properly. This same advice applies to all functions which are used from _lwm2m_registry.c_.

File: subsys/net/lib/lwm2m/lwm2m_registry.c [hunk 2]
```diff
@@ -0,0 +1,1357 @@
...
+#if defined(CONFIG_LWM2M_DTLS_SUPPORT)
+#include <zephyr/net/tls_credentials.h>
+#endif
+#if defined(CONFIG_DNS_RESOLVER)
+#include <zephyr/net/dns_resolve.h>
+#endif
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_rw_link_format.h"
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_registry.c:44]: With a quick look I don't see anything related to content format in this file so all the content format headers can be omitted.

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]
Generally looks good.

**Comment by VeijoPesonen** [comment]
This is up for debate but should we the function name prefixes be changed from _lwm2m_engine__ to _lwm2m_registry__?

File: subsys/net/lib/lwm2m/lwm2m_engine.c (hunk 1 above)
Comment (otkulseng) [reply subsys/net/lib/lwm2m/lwm2m_engine.c:370]: So, every function I've changed from static to non-static should be in the lwm2m_engine.h?

File: subsys/net/lib/lwm2m/lwm2m_registry.c [hunk 3]
```diff
@@ -0,0 +1,1343 @@
...
+#include <zephyr/net/tls_credentials.h>
+#endif
+#if defined(CONFIG_DNS_RESOLVER)
+#include <zephyr/net/dns_resolve.h>
+#endif
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_util.h"
+#if defined(CONFIG_LWM2M_RW_SENML_JSON_SUPPORT)
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_registry.c:39]: My comment about independence from a single content format applies also to the remaining content format headers:
* #include "lwm2m_rw_senml_json.h"
* ...
* #include "lwm2m_rw_senml_cbor.h"

Those shall be removed too.

File: subsys/net/lib/lwm2m/lwm2m_registry.c [hunk 4]
```diff
@@ -0,0 +1,1343 @@
...
+#include <fcntl.h>
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+#if defined(CONFIG_LWM2M_DTLS_SUPPORT)
+#include <zephyr/net/tls_credentials.h>
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_registry.c:30]: You can drop this one.

File: subsys/net/lib/lwm2m/lwm2m_registry.c [hunk 5]
```diff
@@ -0,0 +1,1343 @@
...
+#endif
+#if defined(CONFIG_DNS_RESOLVER)
+#include <zephyr/net/dns_resolve.h>
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_registry.c:33]: You can drop this one too.

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]

**Comment by jheiskan81** [comment]
Looks good for me. It is great that we split massive lwm2m_engine.c file to smaller piece's.

**Comment by otkulseng** [comment]
This was replaced by: #46976

**Comment by rlubos** [comment]
> This was replaced by: #46976

For the future, please don't close existing PRs, especially if they already have comments. You can always extend the scope of the PR.

**Comment by VeijoPesonen** [comment]
> > This was replaced by: #46976
> 
> For the future, please don't close existing PRs, especially if they already have comments. You can always extend the scope of the PR.

My mistake, I gave the advice to replace the existing two with the one.

**Comment by otkulseng** [comment]
> > This was replaced by: #46976
> 
> For the future, please don't close existing PRs, especially if they already have comments. You can always extend the scope of the PR.

Noted

**Pull request closed without merging**